File: src/prod/src/Reliability/Failover/ra/ReplicaOpenMode.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "Ra.Stdafx.h"

using namespace Common;

namespace Reliability
{
    namespace ReplicaOpenMode
    {
        ENUM_STRUCTURED_TRACE(ReplicaOpenMode, Invalid, LastValidEnum);

        void WriteToTextWriter(TextWriter & w, Enum const & val)
        {
            switch (val)
            {
                case Invalid: 
                    w << L"I"; return;
                case New: 
                    w << L"N"; return;
                case Existing: 
                    w << L"E"; return;
                default: 
                    Common::Assert::CodingError("Unknown ReplicaOpenMode value {0}", static_cast<int>(val));
            }
        }

        ::FABRIC_REPLICA_OPEN_MODE ToPublicApi(Enum toConvert)
        {
            switch(toConvert)
            {
            case Invalid:
                return ::FABRIC_REPLICA_OPEN_MODE_INVALID;
            case New:
                return ::FABRIC_REPLICA_OPEN_MODE_NEW;
            case Existing:
                return ::FABRIC_REPLICA_OPEN_MODE_EXISTING;
            default:
                Common::Assert::CodingError("Unknown ReplicaOpenMode value {0}", static_cast<int>(toConvert));
            }
        }
    }
}

File: src/prod/src/managed/Api/src/System/Fabric/ReplicaOpenMode.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric
{
    using System.Fabric.Interop;

    /// <summary>This supports the Service Fabric infrastructure and is not meant to be used directly from your code.</summary>
    public enum ReplicaOpenMode
    {
        /// <summary>This supports the Service Fabric infrastructure and is not meant to be used directly from your code.</summary>
        Invalid = NativeTypes.FABRIC_REPLICA_OPEN_MODE.FABRIC_REPLICA_OPEN_MODE_INVALID,

        /// <summary>This supports the Service Fabric infrastructure and is not meant to be used directly from your code.</summary>
        New = NativeTypes.FABRIC_REPLICA_OPEN_MODE.FABRIC_REPLICA_OPEN_MODE_NEW,

        /// <summary>This supports the Service Fabric infrastructure and is not meant to be used directly from your code.</summary>
        Existing = NativeTypes.FABRIC_REPLICA_OPEN_MODE.FABRIC_REPLICA_OPEN_MODE_EXISTING
    }
}